dlt/cli/_dlt.py

```diff
@@ -304,276 +124,23 @@ def main() -> int:
-            " loaded."
-        ),
-    )
-    pipeline_subparsers.add_parser(
-        "sync",
-        help=(
-            "Drops the local state of the pipeline and resets all the schemas and restores it from"
-            " destination. The destination state, data and schemas are left intact."
-        ),
-        parents=[pipe_cmd_sync_parent],
-    )
-    pipeline_subparsers.add_parser(
-        "trace", help="Displays last run trace, use -v or -vv for more info"
-    )
-    pipe_cmd_schema = pipeline_subparsers.add_parser("schema", help="Displays default schema")
-    pipe_cmd_schema.add_argument(
-        "--format",
-        choices=["json", "yaml"],
-        default="yaml",
-        help="Display schema in this format",
-    )
-    pipe_cmd_schema.add_argument(
-        "--remove-defaults",
-        action="store_true",
-        help="Does not show default hint values",
-        default=True,
-    )
+    # load plugins
+    from dlt.common.configuration import plugins
 
-    pipe_cmd_drop = pipeline_subparsers.add_parser(
-        "drop",
-        help="Selectively drop tables and reset state",
-        parents=[pipe_cmd_sync_parent],
-        epilog=(
-            f"See {DLT_PIPELINE_COMMAND_DOCS_URL}#selectively-drop-tables-and-reset-state for more"
-            " info"
-        ),
-    )
-    pipe_cmd_drop.add_argument(
-        "resources",
-        nargs="*",
-        help=(
-            "One or more resources to drop. Can be exact resource name(s) or regex pattern(s)."
-            " Regex patterns must start with re:"
-        ),
-    )
-    pipe_cmd_drop.add_argument(
-        "--drop-all",
-        action="store_true",
-        default=False,
-        help="Drop all resources found in schema. Supersedes [resources] argument.",
-    )
-    pipe_cmd_drop.add_argument(
-        "--state-paths", nargs="*", help="State keys or json paths to drop", default=()
-    )
-    pipe_cmd_drop.add_argument(
-        "--schema",
-        help="Schema name to drop from (if other than default schema).",
-        dest="schema_name",
-    )
-    pipe_cmd_drop.add_argument(
-        "--state-only",
-        action="store_true",
-        help="Only wipe state for matching resources without dropping tables.",
-        default=False,
-    )
-
-    pipe_cmd_package = pipeline_subparsers.add_parser(
-        "load-package", help="Displays information on load package, use -v or -vv for more info"
-    )
-    pipe_cmd_package.add_argument(
-        "load_id",
-        metavar="load-id",
-        nargs="?",
-        help="Load id of completed or normalized package. Defaults to the most recent package.",
-    )
+    m = plugins.manager()
+    commands = cast(List[Type[SupportsCliCommand]], m.hook.plug_cli())
 
-    subparsers.add_parser("telemetry", help="Shows telemetry status")
+    # install available commands
+    installed_commands: Dict[str, SupportsCliCommand] = {}
+    for c in commands:
```

Comment:
each plugin gets instantiated and registered with the main parser, overwriting of existing commands is prevented. This approach is quite safe, but it will not allow to modify existing parsers or subparsers, this is to be discussed.

Reply:
I'd like to be able to overwrite init command with my own. this prevents that? IMO we should allow rewriting

Reply:
I do not think it is implemented. IMO you should first deduplicate command list, keeping the last occurence. It is better not to add overwritten subparsers to the parser